### PR TITLE
front: use the intervals editor to select the power restrictions on a path

### LINKED ISSUE #4030: Use the intervals editor to POST/PATCH power restrictions
## Description

Use the intervals editor implemented in #4431 to create, update, delete power restrictions on a given path. This step comes in the process of train creation.

Image

## Goal

Ease the train creation process by greatly improving the power restriction edition.

## Acceptance criteria

- [x] The intervals editor implemented in #4431  is used in the `simulation parameters` tab in the train creation window to edit the power restrictions
- [x] The different power restrictions the user must be able to chose from must come from the list sent by this #4020 
- [x] The remarkabke points and begin and end positions of dead sections must be displayed and sapped to when the cursor is close enough. This way the power restrictions set by the user will go from RP to RP or to a dead section, where no power restriction need to be set.
- [x] If the power restriction selected by the user on a given interval is not adapted to the electrification on any part of the interval, a warning must be sent `[The chosen power restriction] is incompatible with electrification [Such] of the path between [begin (m)] and [end (m)]`
- [x] If the user changes either the Rolling Stock or the itinerary, the component must be refreshed, or even hidden if it's not relevant anymore (following the same principle as the one that is currently implemented).
- [x] At the initialization of the component, or after a refresh, it should be pre-filled with empty intervals set at the right positions (begin and end of electrification zones / dead sections)
- [x] If the user goes back and forth between the tabs, without changing neither the rolling stock nor the itinerary, the component must be kept in the same state. Same if the user closes the train creation window and re-opens it (same behavior as the rolling stock, itinerary, etc)

# Questions

- Should we allow empty intervals ? -> Yes, on dead sections it's possible
- What should the warning message be composed of ? Should we sent the position of the conflict ?
- We need to make sure the different power restrictions that have been set are displayed correclty under the Space-Speed Chart

### PR DESCRIPTION
**Pull request by clarani**
closes #4030

### CODE REVIEW

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      };
+    },
+    {}
+  );
+};
+
+interface PowerRestrictionsSelectorProps {
+  rollingStockId: number;
+}
+
```
Comment (alexandredamiron) [inline front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx]: please put this constant after imports

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  const { data: pathWithCatenaries } =
+    osrdEditoastApi.endpoints.getPathfindingByPathIdCatenaries.useQuery(
+      { pathId: pathFindingID as number },
+      { skip: !pathFindingID }
+    );
+
+  const { data: rollingStock } = osrdEditoastApi.endpoints.getRollingStockById.useQuery({
+    id: rollingStockId,
+  });
+
```
Comment (alexandredamiron) [inline front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx:63]: why 1000 ? Also pls put this as a CONST

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx (hunk 1 above)
Comment (clarani) [reply front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx]: Done !

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx (hunk 2 above)
Comment (clarani) [reply front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx:63]: Done !

**Review by alexandredamiron — APPROVED** [review]

**Review by flomonster — APPROVED** [review]
LGTM

**Pull request merged**